src/clj/commiteth/scheduler.clj

```diff
@@ -255,6 +282,43 @@
     (neg? n) (- n)
     :else n))
 
+(defn contract-confirmation-logs [contract-address]
+  "retrives all log events for the confirmation topic since contract creation"
+  (some-> contract-address
+      issues/get-issue-by-contract-address
+      :transaction_hash
```

Comment:
Why are we fetching `transaction_hash` instead of `execute_hash`? (not saying this is wrong, just curious about contract log fetching in general)

Reply:
My feeling was that (at least within the context of the revocation workflow) that the execution_hash would still be in a pending block. 

So transaction_hash is used in this case to define a starting point that we can safely assume has already been mined. Then with the derived block number from that transaction there is enough info to poll the minimal amount of event logs to check for some later event. (in the case of revocation what is actually being polled for is the presence of the execution_hash). 

Thanks for highlighting this as I believe it could be made more extensible by allowing the type of hash (transaction, execution, etc...) to be passed in as a param such as `since_hash` or `starting_point` rather than being hardcoded. Will try and keep that in mind if I spot another use case for this type of polling.